scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,506 @@
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ * ################################################################
+ * $$PROACTIVE_INITIAL_DEV$$
+ */
+package org.ow2.proactive.scheduler.authentication;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.TreeMultimap;
+import org.apache.commons.cli.*;
+import org.apache.commons.io.IOUtils;
+import org.objectweb.proactive.utils.SecurityManagerConfigurator;
+import org.ow2.proactive.authentication.FileLoginModule;
+import org.ow2.proactive.authentication.crypto.CreateCredentials;
+import org.ow2.proactive.authentication.crypto.Credentials;
+import org.ow2.proactive.authentication.crypto.HybridEncryptionUtil;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+import org.ow2.proactive.utils.Tools;
+
+import java.io.*;
+import java.security.KeyException;
+import java.security.PublicKey;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+
+
+/**
+ * Manage user accounts in login.cfg and group.cfg files
+ *
+ * @author The ProActive Team
+ * @see org.ow2.proactive.authentication.crypto.Credentials
+ */
+public class ManageUsers {
+
+    private static final String newline = System.lineSeparator();
+
+    public static final String CREATE_OPTION = "C";
+    public static final String UPDATE_OPTION = "U";
+    public static final String DELETE_OPTION = "D";
+
+    public static final String LOGIN_OPTION = "l";
+    public static final String PWD_OPTION = "p";
+    public static final String GROUPS_OPTION = "g";
+    public static final String KEY_OPTION = "K";
+    public static final String LOGINFILE_OPTION = "L";
+    public static final String GROUPFILE_OPTION = "G";
+
+
+    /**
+     * Entry point
+     *
+     * @param args arguments, try '-h' for help
+     * @throws IOException
+     * @throws ParseException
+     * @see org.ow2.proactive.authentication.crypto.Credentials
+     */
+    public static void main(String[] args) {
+
+        try {
+
+            manageUsers(args);
+
+        } catch (ManageUsersException e) {
+            System.err.println("ERROR : " + e.getMessage());
+            if (e.getCause() != null) {
+                System.err.println(e.getCause().getMessage());
+            }
+            if (e.getAdditionalInfo() != null) {
+                System.out.println(e.getAdditionalInfo());
+            }
+            System.exit(1);
+        }
+
+        System.exit(0);
+    }
+
+    public static void manageUsers(String[] args) throws ManageUsersException {
```

Comment:
this method is a bit too complex. Could you refactor externalising some logic to other methods or classes
?